### PR TITLE
Fixed image path search

### PR DESCRIPTION
**Pull request by Clonkk**
@pietroppeter I've thought about https://github.com/pietroppeter/nimib/issues/54. 

I think this should do the trick. It seem to work locally for scinim example. 

I made a distinction between : 
* relative path and  that should used nbDoc.context 
* absolute path that should used without modification.

What do you think ?

### CODE REVIEW

**Comment by pietroppeter** [comment]
yes, I think this could work, while we wait for #55 (the plan then should be to use `nbRelUrl` mentioned there where the implementation could be based directly on `nbSrcDir` but it would likely still use some sort of isAbsolute mechanism).

I also checked and `isAbsolute` works also for real urls (e.g. "http://..." is declared NOT absolute).

I can merge and create a new release, ok?

Thanks!

**Comment by Clonkk** [comment]
> I also checked and isAbsolute works also for real urls (e.g. "http://..." is declared NOT absolute).

Should that be handled as a separate case then ? non-absolute URL gets added a prefix.

> I can merge and create a new release, ok?

I don't know why the CI fails, but if that doesn't bother you, sure :)

**Comment by Clonkk** [comment]
I ended up making a special case for url that starts with http.

It's quick and dirty but should work for the time being until your refactoring #55 is done :)

**Comment by HugoGranstrom** [comment]
The error in CI is this:
```
:/home/runner/work/nimib/nimib/docs/penguins.nim(37, 8) template/generic instantiation of `nbImage` from here
/home/runner/work/nimib/nimib/src/nimib.nim(70, 11) Error: undeclared field: 'isAbsolute' for type system.string [declared in /home/runner/.choosenim/toolchains/nim-1.4.4/lib/system.nim(34, 3)] 
```
Which is odd because `isAbsolute` is defined in `os` ([here](https://nim-lang.org/docs/os.html#isAbsolute%2Cstring)) and it is imported at the top of `src/nimib.nim`. I think I might have stumbled upon something similar when I tried to get `-d:nimibHomeDir` to work. I think I did `bla.blabla.absoluteThing` and go errors but when I changed it to `absoluteThing(bla.blabla)` it worked. Might be worth a try?

**Comment by Clonkk** [comment]
The one you pointed out was the first error that then got fixed in the second commit :).
I tried it anyway (you never know with CI...) but that wasn't it :D

The current error is : 
```
Error: execution of an external program failed: '/home/runner/.cache/nim/ptest_d/ptest '
stack trace: (most recent call last)
/tmp/nimblecache-946033505/nimscriptapi_905765425.nim(187, 16)
/home/runner/work/nimib/nimib/nimib.nimble(23, 8) ptestTask
/home/runner/.choosenim/toolchains/nim-1.4.4/lib/system/nimscript.nim(260, 7) exec
/home/runner/.choosenim/toolchains/nim-1.4.4/lib/system/nimscript.nim(260, 7) Error: unhandled exception: FAILED: nim r docs/ptest [OSError]
     Error: Exception raised during nimble script execution
Error: Process completed with exit code 1.
```
which I don't know how to fix.

**Comment by HugoGranstrom** [comment]
Hmmm okay. That error doesn't give us any clues at all :/ Have you run the tests locally and gotten any better error messages? (I'm on my way to test now)

**Comment by HugoGranstrom** [comment]
Oh wait, `ptest` wasn't just a regular test that just compiles and if it does it passes. It compares the generated HTML with the one already existing and if they differ it fails *on exit*, hence no concrete error messages?

**Comment by pietroppeter** [comment]
The error in CI is not nicely visibile but parsing the content I can see that only penguins is failing and the reasons are diffs like this: https://github.com/pietroppeter/nimib/runs/2787066262#step:6:230
where you see it is complaining that `<img src="./images/penguins_count_species.png"` is changed to `<img src="images/penguins_count_species.png"`.

The fix is to run penguins locally and commit the output, being careful not to commit line with the different github repo because of this #41.

File: docs/index.html [hunk 1]
```diff
@@ -259,7 +259,7 @@ <h2>🙏 Thanks</h2>
 <p>to:</p>
 <ul>
 <li><a href="https://github.com/soasme">soasme</a> for the excellent libraries nim-markdown and nim-mustache, which provide the backbone of nimib rendering and customization</li>
-<li><a href="https://github.com/clonkk">clonkk</a> for help in a critical piece of code early on (see <a href="https://stackoverflow.com/a/64032172/4178189">this Stack Overflow answer</a>)</li>
+<li><a href="https://github.com/pietroppeter">pietroppeter</a> for help in a critical piece of code early on (see <a href="https://stackoverflow.com/a/64032172/4178189">this Stack Overflow answer</a>)</li>
```
Comment (HugoGranstrom) [inline docs/index.html:262]: Are you removing yourself from the credits? 😛

Comment (Clonkk) [reply docs/index.html:262]: I just used sed on the hole repo lol

File: docs/cheatsheet.html [hunk 2]
```diff
@@ -37,7 +37,7 @@
...
+  <span><a href="https://github.com/Clonkk/nimib"><svg aria-hidden="true" width="1.2em" height="1.2em" style="vertical-align: middle;" preserveAspectRatio="xMidYMid meet" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59c.4.07.55-.17.55-.38c0-.19-.01-.82-.01-1.49c-2.01.37-2.53-.49-2.69-.94c-.09-.23-.48-.94-.82-1.13c-.28-.15-.68-.52-.01-.53c.63-.01 1.08.58 1.23.82c.72 1.21 1.87.87 2.33.66c.07-.52.28-.87.51-1.07c-1.78-.2-3.64-.89-3.64-3.95c0-.87.31-1.59.82-2.15c-.08-.2-.36-1.02.08-2.12c0 0 .67-.21 2.2.82c.64-.18 1.32-.27 2-.27c.68 0 1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82c.44 1.1.16 1.92.08 2.12c.51.56.82 1.27.82 2.15c0 3.07-1.87 3.75-3.65 3.95c.29.25.54.73.54 1.48c0 1.07-.01 1.93-.01 2.2c0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z" fill="#000"></path></svg></a></span>
```
Comment (HugoGranstrom) [inline docs/cheatsheet.html:40]: I think you overdid it :)

File: docs/ptest.html [hunk 3]
```diff
@@ -35,7 +35,7 @@
...
+  <span><a href="https://github.com/Clonkk/nimib"><svg aria-hidden="true" width="1.2em" height="1.2em" style="vertical-align: middle;" preserveAspectRatio="xMidYMid meet" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59c.4.07.55-.17.55-.38c0-.19-.01-.82-.01-1.49c-2.01.37-2.53-.49-2.69-.94c-.09-.23-.48-.94-.82-1.13c-.28-.15-.68-.52-.01-.53c.63-.01 1.08.58 1.23.82c.72 1.21 1.87.87 2.33.66c.07-.52.28-.87.51-1.07c-1.78-.2-3.64-.89-3.64-3.95c0-.87.31-1.59.82-2.15c-.08-.2-.36-1.02.08-2.12c0 0 .67-.21 2.2.82c.64-.18 1.32-.27 2-.27c.68 0 1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82c.44 1.1.16 1.92.08 2.12c.51.56.82 1.27.82 2.15c0 3.07-1.87 3.75-3.65 3.95c.29.25.54.73.54 1.48c0 1.07-.01 1.93-.01 2.2c0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z" fill="#000"></path></svg></a></span>
```
Comment (HugoGranstrom) [inline docs/ptest.html:38]: This one and in `nolan_no_source.html` it's still incorrect. I guess this one specifically might be the one triggering the CI atm

**Comment by Clonkk** [comment]
I removed whitespace consideration in the git diff command. Hopefully that should make the CI pass

**Comment by HugoGranstrom** [comment]
Woop woop! 😁 Now I have no excuses to not start writing some content 😛

**Comment by pietroppeter** [comment]
ok, looks all good to me. thanks for this and also for the git diff fix!
I will merge and release now.

**Comment by HugoGranstrom** [comment]
Great! Rerunning the deployment now

**Pull request merged**